acme/acme/crypto_util.py

```diff
@@ -0,0 +1,146 @@
+"""Crypto utilities."""
+import contextlib
+import logging
+import socket
+import sys
+
+from six.moves import range  # pylint: disable=import-error,redefined-builtin
+
+import OpenSSL
+
+from acme import errors
+
+
+logger = logging.getLogger(__name__)
+
+
+_DEFAULT_SSL_METHOD = OpenSSL.SSL.SSLv23_METHOD
```

Comment:
There are soooo many major problems with SSL.  It should be removed from servers altogether.

We should be using OpenSSL.SSL.TLSv1_METHOD at a minimum.  TLSv1_2 would probably work for our purposes.  We should be dealing with modern TLS clients.